content/docs/rfcs/44/README.md

```diff
@@ -0,0 +1,275 @@
+
+## State Flag
+
+44/WAKU2-DANDELION extends [Waku Relay](/spec/11/#wire-specification),
+and [libp2p pubsub](https://github.com/libp2p/specs/tree/master/pubsub#the-message),
+by a `stem` flag, which indicates whether a message is in the stem phase or not.
+We refer to the former as a stem message and to the latter as a fluff message.
+
+```protobuf
+syntax = "proto2";
+
+message RPC {
+  repeated SubOpts subscriptions = 1;
+  repeated Message publish = 2;
+
+  message SubOpts {
+    optional bool subscribe = 1;
+    optional string topicid = 2;
+  }
+
+  message Message {
+    optional string from = 1;
+    optional bytes data = 2;
+    optional bytes seqno = 3;
+    repeated string topicIDs = 4;
+    optional bytes signature = 5;
+    optional bytes key = 6;
+    optional bool stem = 7;
+  }
+}
+```
+
+## Stem State
+
+On entering stem phase,
+nodes supporting 44/WAKU2-DANDELION MUST randomly select two nodes for each pubsub topic from the respective gossipsub mesh node set.
+These nodes are referred to as stem relays. (These stem relays themselves can either be in stem state or in fluff state.)
+Further, the node establishes a map that maps each incoming stem connection
+to one of its stem relays chosen at random (but fixed per epoch).
+Incoming stem connections are identified by the stem flag in messages received on the respective connection.
+The node itself is mapped in the same way, so that all messages originated by the node are relayed via a per-epoch-fixed Dandelion relay, too.
+
+While in stem state, nodes MUST relay stem messages to the respective stem relay.
+While in stem state, nodes MUST NOT gossip about stem messages,
+and MUST NOT send control messages related to stem messages.
+Stem messages relayed in stem state MUST have the stem flag set.
+Fluff messages MUST be relayed as specified in the fluff state section.
+
+### Fail Safe
+
+Nodes $v$ in stem state SHOULD store messages attached with a random timer between $t_1 = 5 * 100ms$ and $t_2 = 2 * t_1$.
+This time interval is chosen because
+
+* we assume $100\,ms$ as an average per hop delay, and
+* using $q=0.2$ will lead to an expected number of 5 stem hops per message.
+
+If $v$ does not receive a given message via Waku relay (fluff) before the respective timer runs out,
+$v$ will disseminate the message via Waku relay.
+
+## Fluff State
+
+In fluff state, nodes operate as usual Waku Relay nodes, with the exception of adding a random delay between $0$ and $100\,ms$ before relaying a message.
+Before relaying a message in fluff state, the `stem` flag MUST be set to false.
+
+As mentioned in previous sections, we might move the stem phase specification into a separate document.
+We will look more into (optimal) delay ranges and might introduce dynamic delay ranges based on traffic volume.
+
+# Implementation Notes
+
+Since 44/WAKU2-DANDELION alters message dissemination,
+it is implemented on the libp2p gossipsub layer.
```

Comment:
Maybe it would be better to do the stem phase over light push, and then fluff over GS? This way LP is also protected (I think), and no need to change the core of GS